gateways/cardconnect/cardconnect.go

```diff
@@ -0,0 +1,161 @@
+package cardconnect
+
+import (
+	"bytes"
+	"encoding/base64"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/BoltApp/sleet"
+	"github.com/BoltApp/sleet/common"
+)
+
+func NewClient(username string, password string, merchantID string, URL string, environment common.Environment) *CardConnectClient {
+	return NewWithHttpClient(username, password, merchantID, URL, environment, common.DefaultHttpClient())
+}
+
+// NewWithHttpClient uses authentication with custom http client
+func NewWithHttpClient(username string, password string, merchantID string, URL string, environment common.Environment, httpClient *http.Client) *CardConnectClient {
+	return &CardConnectClient{
+		httpClient: httpClient,
+		username:   username,
+		password:   password,
+		merchantID: merchantID,
+		URL:        URL,
```

Comment:
is the URL custom per user? so each merchant will need to provide their own URL?